src/Card.js

```diff
@@ -75,25 +80,72 @@ export default class Card extends Phaser.GameObjects.Container {
   drag (ptr, dragX, dragY) {
     if(!this.playerPickedUp) return
     this.setDepth(activeDepth)
-    const { dragHistory } = this
-    dragHistory.push([dragX, dragY]);
-    if(dragHistory.length > 2) {
-      const [lastX, lastY] = dragHistory[dragHistory.length - 1];
-      const [penX, penY] = dragHistory[dragHistory.length - 2];
-      const dx = (lastX - penX) * 50;
-      const dy = (lastY - penY) * 50;
-      this.body.setVelocity(dx, dy);
-    } else this.body.setVelocity(0,0)
-    this.x = dragX;
-    this.y = dragY;
+    if (this.inHand) {
+      // console.log('dragX', dragX, 'dragY', dragY);
+
+      switch(this.playerNumber) {
+        case 2:
+          this.y = dragY < 0 ? 0 : dragY > 800 ? 800 : dragY;
+          this.x = canvasWidth + inHandAdjustment;
+          break;
+        case (3):
+          this.x = dragX < 0 ? 0 : dragX > 800 ? 800 : dragX;
+          this.y = 0 - inHandAdjustment;
+          break;
+        case 4:
+          this.y = dragY < 0 ? 0 : dragY > 800 ? 800 : dragY;
+          this.x = 0 - inHandAdjustment;
+          break;
+        default:
+          this.x = dragX < 0 ? 0 : dragX > 800 ? 800 : dragX;
+          this.y = canvasHeight + inHandAdjustment;
+          break;
+      }
+
+      if (!this.addToHand) {
+        this.inHand = false;
+        this.body.setCollideWorldBounds(true);
+        this.gameState.cards[this.cardNumber] = this;
+        delete this.gameState.hands[`player${this.playerNumber}`][this.cardNumber];
+
+        this.socket.emit('removeCardFromHand', {cardNumber: this.cardNumber, room: this.gameState.room, player: `player${this.playerNumber}`});
+      }
+    } else {
+      const { dragHistory } = this
+      dragHistory.push([dragX, dragY]);
+      if(dragHistory.length > 2 && !this.addToHand) {
+        const [lastX, lastY] = dragHistory[dragHistory.length - 1];
+        const [penX, penY] = dragHistory[dragHistory.length - 2];
+        const dx = (lastX - penX) * 50;
+        const dy = (lastY - penY) * 50;
+        this.body.setVelocity(dx, dy);
+      } else this.body.setVelocity(0,0)
+      this.x = dragX;
+      this.y = dragY;
+    }
+
+    switch(this.playerNumber) {
+      case 2:
+        this.addToHand = dragX > canvasWidth - 100;
```

Comment:
maybe should have 100 in constants so we can change dimensions if we change table size